examples/cpp-simple/server.cc

```diff
@@ -16,101 +16,115 @@
 // This example really does nothing other than show how to integrate
 // the C++ SDK.
 
-#include <iostream>
-#include <thread>
 #include <agones/sdk.h>
 #include <grpc++/grpc++.h>
+#include <iostream>
+#include <thread>
 
 // send health check pings
 void doHealth(agones::SDK *sdk) {
-    while (true) {
-        if (!sdk->Health()) {
-            std::cout << "Health ping failed" << std::endl;
-        } else {
-            std::cout << "Health ping sent" << std::endl;
-        }
-        std::this_thread::sleep_for(std::chrono::seconds(2));
+  while (true) {
+    if (!sdk->Health()) {
+      std::cout << "Health ping failed" << std::endl;
+    } else {
+      std::cout << "Health ping sent" << std::endl;
     }
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+  }
 }
 
 // watch GameServer Updates
 void watchUpdates(agones::SDK *sdk) {
-    std::cout << "Starting to watch GameServer updates..." << std::endl;
-    sdk->WatchGameServer([](stable::agones::dev::sdk::GameServer gameserver){
-        std::cout << "GameServer Update, name: " << gameserver.object_meta().name() << std::endl;
-        std::cout << "GameServer Update, state: " << gameserver.status().state() << std::endl;
-    });
+  std::cout << "Starting to watch GameServer updates..." << std::endl;
+  sdk->WatchGameServer([](agones::dev::sdk::GameServer gameserver) {
+    std::cout << "GameServer Update, name: " << gameserver.object_meta().name()
+              << std::endl;
+    std::cout << "GameServer Update, state: " << gameserver.status().state()
+              << std::endl;
+  });
 }
 
 int main() {
-    std::cout << "C++ Game Server has started!" << std::endl;
-
-    std::cout << "Getting the instance of the SDK!" << std::endl;
-    agones::SDK *sdk = new agones::SDK();
-
-    std::cout << "Attempting to connect..." << std::endl;
-    bool connected = sdk->Connect();
-    if (!connected) {
-        std::cout << "Could not connect to the sidecar. Exiting!" << std::endl;
-        return -1;
-    }
-    std::cout << "...handshake complete." << std::endl;
-
-    std::thread health (doHealth, sdk);
-    std::thread watch (watchUpdates, sdk);
-
-    std::cout << "Setting a label" << std::endl;
-    grpc::Status status = sdk->SetLabel("test-label", "test-value");
-    if (!status.ok()) {
-        std::cout << "Could not run SetLabel(): "+ status.error_message() + ". Exiting!" << std::endl;
-        return -1;
-    }
-
-    std::cout << "Setting an annotation" << std::endl;
-    status = sdk->SetAnnotation("test-annotation", "test value");
-    if (!status.ok()) {
-        std::cout << "Could not run SetAnnotation(): "+ status.error_message() + ". Exiting!" << std::endl;
+  std::cout << "C++ Game Server has started!" << std::endl;
+
+  std::cout << "Getting the instance of the SDK!" << std::endl;
+  agones::SDK *sdk = new agones::SDK();
```

Comment:
I don't think there's any need to heap-allocate the SDK. I recommend stack allocating it like `agones::SDK sdk;`

Reply:
done.

Reply:
So after doing this I ran into an issue where I need to pass the sdk as a pointer into the two functions that run in separate threads. If I try to pass by value/reference, I get this compile error:

```
In file included from /project/cpp-simple/server.cc:22:
/usr/local/include/c++/8.3.0/thread: In instantiation of 'std::thread::thread(_Callable&&, _Args&& ...) [with _Callable = void (&)(agones::SDK*); _Args = {agones::SDK&}; <template-parameter-1-3> = void]':
/project/cpp-simple/server.cc:61:35:   required from here
/usr/local/include/c++/8.3.0/thread:120:17: error: static assertion failed: std::thread arguments must be invocable after conversion to rvalues
  static_assert( __is_invocable<typename decay<_Callable>::type,
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           typename decay<_Args>::type...>::value,
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/c++/8.3.0/thread: In instantiation of 'struct std::thread::_Invoker<std::tuple<void (*)(agones::SDK*), agones::SDK> >':
/usr/local/include/c++/8.3.0/thread:132:22:   required from 'std::thread::thread(_Callable&&, _Args&& ...) [with _Callable = void (&)(agones::SDK*); _Args = {agones::SDK&}; <template-parameter-1-3> = void]'
/project/cpp-simple/server.cc:61:35:   required from here
/usr/local/include/c++/8.3.0/thread:250:2: error: no matching function for call to 'std::thread::_Invoker<std::tuple<void (*)(agones::SDK*), agones::SDK> >::_M_invoke(std::thread::_Invoker<std::tuple<void (*)(agones::SDK*), agones::SDK> >::_Indices)'
  operator()()
  ^~~~~~~~
...
```

Now scoping the lifetime of the sdk to the main function makes less sense, since it's being shared by the main thread, the health checking thread, and the thread watching for updates to the gameserver definition. If this were a regular program and this was a function, you'd deallocate the object while retaining a reference and end up in a bad state (since this is main, the program exits so it doesn't really matter). 

What do you think about leaving it this way vs. putting it back to heap allocation? I realize that the `delete sdk` call often isn't called (if the function has any non-success exit) but how much should we care?